assets/src/blocks/Happypoint/HappypointFrontend.js

```diff
@@ -0,0 +1,108 @@
+import { Component } from '@wordpress/element';
+
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+export class HappypointFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {}
+    };
+
+    this.getData = this.getData.bind(this);
+  };
+
+  componentDidMount() {
+    this.getData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { id, iframe_url, mailing_list_iframe } = this.props;
+    if (
```

Comment:
Not a blocker, but if we are going to use hooks more and more, this looks like it could be expressed better as an effect hook with dependencies ( ìd`, `iframe_url`, `mailing_list_iframe`). Maybe it wouldn't cost too much to turn this class into a functional component and let this comparisons happen automatically through memoization.